src/lib/optimizer/strategy/like_replacement_rule.cpp

```diff
@@ -0,0 +1,87 @@
+#include "like_replacement_rule.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+#include "all_parameter_variant.hpp"
+#include "constant_mappings.hpp"
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "operators/operator_scan_predicate.hpp"
+#include "statistics/chunk_statistics/chunk_statistics.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/assert.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+std::string LikeReplacementRule::name() const { return "Like Replacement Rule"; }
+
+void LikeReplacementRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) const {
+  const auto predicate_node = std::dynamic_pointer_cast<PredicateNode>(node);
+  // Continue only on predicate nodes
+  if (predicate_node == nullptr) {
+    _apply_to_inputs(node);
+    return;
+  }
+  const auto expression = std::dynamic_pointer_cast<BinaryPredicateExpression>(predicate_node->predicate());
+  // Continue only if expressions predicate condition is of type PredicateCondition::Like
+  if (expression == nullptr || expression->predicate_condition != PredicateCondition::Like) {
+    _apply_to_inputs(node);
+    return;
+  }
+  const auto column_expression = std::dynamic_pointer_cast<LQPColumnExpression>(expression->left_operand());
```

Comment:
the left operand being a column expression is by no means a necessity - please allow any expression type here.

Reply:
Mmh, not sure if I agree. While the transformation would still be valid, it might not be beneficial.

For now, I'd keep it simple and stick to the default case of having a column on the left.

Reply:
For which expression `x` we want `x LIKE "a%"` over `x BETWEEN a AND b (EX)`?

Reply:
Mmh, now that I think about it a bit longer, I can't come up with anything. Yes, I suppose we can simply get rid of the check for a ColumnExpression on the left.